tools/assets/verify_issue.ts

```diff
@@ -0,0 +1,68 @@
+import {spawnSync} from 'child_process';
+import {readFile, writeFile} from 'fs/promises';
+
+(async () => {
+  const error = await readFile('puppeteer-error.txt', 'utf-8');
+  const behavior = JSON.parse(
+    await readFile('puppeteer-behavior.json', 'utf-8')
+  ) as {flaky?: boolean; noError?: boolean};
+
+  let maxRepetitions = 1;
+  if (behavior.flaky) {
+    maxRepetitions = 100;
+  }
+
+  let status: number | null = null;
+  let stderr = '';
+  let stdout = '';
+
+  const preHook = async () => {
+    console.log('Writing output and error logs...');
+    await Promise.all([
+      writeFile('output.log', stdout),
+      writeFile('error.log', stderr),
+    ]);
+  };
+
+  let checkStatusWithError: () => Promise<void>;
+  if (behavior.noError) {
+    checkStatusWithError = async () => {
+      if (status === 0) {
+        await preHook();
+        console.log('Script ran successfully; no error found.');
+        process.exit(0);
+      }
+    };
+  } else {
+    checkStatusWithError = async () => {
+      if (status !== 0) {
+        await preHook();
+        if (stderr.toLowerCase().includes(error.toLowerCase())) {
+          console.log('Script failed; error found.');
+          process.exit(0);
+        }
+        console.error('Script failed; unknown error found.');
+        process.exit(1);
+      }
+    };
+  }
+
+  for (let i = 0; i < maxRepetitions; ++i) {
+    const result = spawnSync('npm', ['start'], {
```

Comment:
do we want a timeout here if the user script never resolves?

Reply:
I put it on the step.